include/core/platform/lf_arduino_support.h

```diff
@@ -90,6 +90,19 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #endif
 #endif
 
+#define __STDC_FORMAT_MACROS
+#include <inttypes.h>
+
+#ifndef _SYS__TIMESPEC_H_
+#define	_SYS__TIMESPEC_H_
+
+struct timespec {
+	long long	tv_sec;		/* seconds */
+	long	tv_nsec;	/* and nanoseconds */
+};

```

Comment:
What is this used for?

Reply:
AVR doesn't have timespec to hold the time when you first load the program. MBED does, so I just made the custom timespec struct to hold it.

Reply:
Hm. I don't understand what you mean. Is there some external dependency we are linking with that needs this struct defined?

Reply:
I'm also a bit puzzled by this code. Do we actually need it?

Reply:
Inside of reactor_common.c, when we initialize we have the following set of code:
```
LF_PRINT_DEBUG("Start time: " PRINTF_TIME "ns", start_time);
struct timespec physical_time_timespec = {physical_start_time / BILLION, physical_start_time % BILLION};
printf("---- Start execution at time %s---- plus %ld nanoseconds.\n",
            ctime(&physical_time_timespec.tv_sec), physical_time_timespec.tv_nsec);
```

This is the only place the timespec is referenced for the Arduino lifecycle, but I myself don't know what purpose this has.

Reply:
For MBED systems like the BLE Sense Boards, timespec is already defined, but AVR doesn't have it formally defined. To prevent the system from breaking, I simply hardcoded the timespec struct. That said, I still don't know the underlying purpose of the struct if we simply read to and from the struct and have it get discarded later?

Reply:
It looks like the federated reactor still needs timespec and I'm currently working on federated support for Arduino so I think it's best to keep it.